velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -52,13 +52,66 @@ struct ArrayMinMaxFunction {
     }
   }
 
+  template <typename TReturn, typename TInput>
+  bool callForFloatOrDouble(TReturn& out, const TInput& array) {
+    bool hasNull = false;
+    auto it = array.begin();
+
+    // Find the first non-null item (if any)
+    while (it != array.end()) {
```

Comment:
Is this loop shared with the existing implementation? Consider extracting a helper function to avoid copy-paste.

Reply:
This is not shared with existing implementation. The reasoning is - to find min (or max), we need a reference point and compare with elements as we proceed forward in the iteration. The reference point is the first non-null value. The case is simple for integers or strings (where NAN is not present), because first item can be either NULL or a value. So current implementation just looks at the first item, and keep the reference point or returns.

```
    auto it = array.begin();
    // Result is null if any element is null.
    if (!it->has_value()) {
      return false;
    }

   auto currentValue = it->value();
```

Now for float or double array - the first item can be NULL, NAN or a value. Now if its NAN, we can return early like non-float case. If it's NULL, we can't return and we don't have the reference point yet. So the iteration looks further in the array if there is a non null value that can be used as reference value (`currentValue` in code)

```
    // Find the first non-null item (if any)
    while (it != array.end()) {
      if (it->has_value()) {
        break;
      } else {
        hasNull = true;
      }
      ++it;
    }

    // If first non-null item is NAN, return immediately.
    auto currentValue = it->value();
```

Since the logic is separate, I kept the existing implementation as is. We utilize existing functions where possible (e.g. update, assign)

Reply:
Thank you for clarifying. Makes sense.